docs/design/data_plane_manager.adoc

```diff
@@ -0,0 +1,1189 @@
+____
+* {blank}
++
+____
+Request: 
+____
+
+____
+"/goalstate/{goalstateid}","/v4/goalstate/{goalstateid}"
+____
+
+* {blank}
++
+____
+Request Parameter: 
+____
+** @ PathVariable String portid
+* {blank}
++
+____
+Response: List of GoalState Information
+____
+* {blank}
++
+____
+Example
+____
+
+Request:
+
+http://serverIP:8080/goalstate/f37810eb-7f83-45fa-a4d4-1b31e75399df
+
+Response:
+
+[source,c++]
+------------------------------------------------------------
+{
+  "goalstate": [{
+    "vpc": {
+      "project_id": "3dda2801-d675-4688-a63f-dcda8d327f50",
+      "id": "9192a4d4-ffff-4ece-b3f0-8d36e3d88038",
+      "name": "test_vpc",
+      "description": "",
+      "cidr": "10.0.0.0/16"
+    },
+    "subnet": {
+      "project_id": "3dda2801-d675-4688-a63f-dcda8d327f50",
+      "vpc_id": "9192a4d4-ffff-4ece-b3f0-8d36e3d88038",
+      "id": "a87e0f87-a2d9-44ef-9194-9a62f178594e",
+      "name": "test_subnet",
+      "description": "",
+      "cidr": "10.0.0.0/20",
+      "gateway_ip": "10.0.0.5",
+      "availability_zone": "uswest-1",
+      "dhcp_enable": false,
+      "primary_dns": null,
+      "secondary_dns": null,
+      "dns_list": null
+    },
+    "port": {
+      "project_id": "3dda2801-d675-4688-a63f-dcda8d327f50",
+      "id": "f37810eb-7f83-45fa-a4d4-1b31e75399df",
+      "name": "test_cni_port2",
+      "description": "",
+      "network_id": "a87e0f87-a2d9-44ef-9194-9a62f178594e",
+      "tenant_id": null,
+      "admin_state_up": true,
+      "mac_address": null,
+      "veth_name": "veth0",
+      "device_id": null,
+      "device_owner": null,
+      "status": null,
+      "fixed_ips": [
+        
+      ],
+      "allowed_address_pairs": null,
+      "extra_dhcp_opts": null,
+      "security_groups": null,
+      "binding:host_id": "ephost_0",
+      "binding:profile": null,
+      "binding:vnic_type": null,
+      "network_ns": "/var/run/netns/test_netw_ns",
+      "dnsName": null,
+      "dnsAssignment": null,
+      "fast_path": true
+    },
+    "security_group": {
+      "project_id": "3dda2801-d675-4688-a63f-dcda8d327f50",
+      "version": 0,
+      "vpc_id": "9192a4d4-ffff-4ece-b3f0-8d36e3d88038",
+      "id": "f37810eb-7f83-45fa-a4d4-1b31e75399df",
+      "name": "test_security_group1",
+      "security_group_rules": null
+    }
+  }]
+}
+------------------------------------------------------------
+
+anchor:Create_Port_GoalState[]
+
+====  (2) Send GoalState to create port
+
+* {blank}
++
+____
+Method: POST
+____
+* {blank}
++
+____
+Request: "/goalstate/{gsinfo}", "/v4/goalstate/{gsinfo}"
+____
+* {blank}
++
+____
+Request Parameter: 
+____
+** @RequestBody GSInfo resource
+* {blank}
++
+____
+Response: port creation result information
+____
+* {blank}
+* Normal response codes: 200
+* Error response codes: 400, 412, 500
++
+____
+Example
+____
+
+Request:
+
+http://localhost:8080/goalstate/
+
+Body:
+
+------------------------------------------------------------
+{
+  "operation_type": "create",
```

Comment:
We don't need operation_type for DPM APIs, as the API methods (POST/PUT/DELETE/GET) include the information already.